druid/src/core.rs

```diff
@@ -558,159 +561,168 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
             return;
         }
         let had_active = self.state.has_active;
-        let mut child_ctx = EventCtx {
-            cursor: ctx.cursor,
-            state: ctx.state,
-            widget_state: &mut self.state,
-            is_handled: false,
-            is_root: false,
-        };
+        let rect = self.state.layout_rect.unwrap_or_default();
 
-        let rect = child_ctx.widget_state.layout_rect.unwrap_or_default();
+        // If we need to replace either the event or its data.
+        let mut modified_event = None;
+        // Tracking whether we need to merge our state up to the parent.
+        let mut needs_merge = false;
```

Comment:
is this an optimization (my assumption), or are there times when merging would be incorrect?

if an optimization: my gut feeling is that merging is only a trivial amount of work in the general case, such that this optimization is unlikely to be meaningful even over very large graphs, and I'm not sure it's something it is worth adding more logic to handle, at least without being guided by a concrete performance concern?